lib/workers/repository/update/branch/bump-versions.spec.ts

```diff
@@ -0,0 +1,349 @@
+          {
+            type: 'addition',
+            path: 'foo',
+            contents: 'bar',
+          },
+        ],
+        updatedArtifacts: [],
+      });
+      scm.getFileList.mockResolvedValueOnce(['foo']);
+
+      await bumpVersions(config);
+
+      expect(config).toMatchObject({
+        artifactErrors: [
+          {
+            stderr:
+              "Failed to compile matchString for bumpVersions: Unexpected token '{'",
+          },
+        ],
+      });
+    });
+
+    it('should catch template error in bumpType', async () => {
+      const compile = vi.spyOn(templates, 'compile');
+      compile.mockReturnValueOnce('foo');
+      compile.mockReturnValueOnce('^(?<version>.+)$');
+      compile.mockImplementationOnce(() => {
+        throw new Error("Unexpected token '{'");
+      });
+      const config = partial<BranchConfig>({
+        bumpVersions: [
+          {
+            fileMatch: ['foo'],
+            bumpType: 'minor',
+            matchStrings: ['^(?<version>.+)$'],
+          },
+        ],
+        updatedPackageFiles: [
+          {
+            type: 'addition',
+            path: 'foo',
+            contents: '1.0.0',
+          },
+        ],
+        updatedArtifacts: [],
+      });
+      scm.getFileList.mockResolvedValueOnce(['foo']);
+
+      await bumpVersions(config);
+
+      expect(config).toMatchObject({
+        artifactErrors: [
+          {
+            stderr:
+              "Failed to calculate new version for bumpVersions: Unexpected token '{'",
+          },
+        ],
+      });
+    });
+
+    it('should bump version in a non edited file and add to updatedArtifacts', async () => {
+      const config = partial<BranchConfig>({
+        bumpVersions: [
+          {
+            fileMatch: ['\\.release-version'],
+            bumpType: 'minor',
+            matchStrings: ['^(?<version>.+)$'],
+          },
+        ],
+        updatedPackageFiles: [
+          {
+            type: 'addition',
+            path: 'foo',
+            contents: 'bar',
+          },
+        ],
+      });
+      scm.getFileList.mockResolvedValueOnce(['foo', '.release-version']);
+      fs.readLocalFile.mockResolvedValueOnce('1.0.0');
+      await bumpVersions(config);
+
+      expect(config).toMatchObject({
+        updatedArtifacts: [
+          {
+            type: 'addition',
+            path: '.release-version',
+            contents: '1.1.0',
+          },
+        ],
+      });
+    });
+
+    it('should bump version in an already changed packageFiles', async () => {
+      const config = partial<BranchConfig>({
+        bumpVersions: [
+          {
+            fileMatch: ['foo'],
+            bumpType: 'minor',
+            matchStrings: ['\\s*version:\\s+(?<version>[^\\s]+)'],
+          },
+        ],
+        updatedPackageFiles: [
+          {
+            type: 'addition',
+            path: 'foo',
+            contents: codeBlock`
+            version: 1.0.0
+            dependencies: {}
```

Comment:
```suggestion
              version: 1.0.0
              dependencies: {}
```

i would indent it two more spaces for better visibillity

Reply:
☝️ ❓ Please leave a comment why you not accept my proposals 🙏